core/trino-main/src/main/java/io/trino/sql/analyzer/Analysis.java

```diff
@@ -1058,14 +1066,17 @@ public boolean hasRowFilter(QualifiedObjectName table, String identity)
         return rowFilterScopes.contains(new RowFilterScopeEntry(table, identity));
     }
 
-    public void registerTableForRowFiltering(QualifiedObjectName table, String identity)
+    public void registerTableForRowFiltering(QualifiedObjectName table, String identity, String filterExpression)
```

Comment:
TableInfo already stores `filterExpression` - what is the difference between this reference and the `TableInfo#getFilters` ?

Reply:
Yeah let me address this question and also the top-level one you posed

> I'm not sure if we need to capture the row filter and column masks along with this - as we already have them in TableInfo#getFilters and TableInfo#getColumns where we maintain each information about the column.

The `filters` and `columns` captured in `TableInfo` record the filters/masks applied to a given table. In contrast, the filters and masks stored in `ViewLikeInfo`/`referenceChain` record the filters/masks that _reference_ a table. Let's take an example to make this clear.

Say we have the query below (assume default catalog `default` and default schema `db`):
```
SELECT a, b FROM t_base;
```

There is a row filter applied to `t_base` with the expression `a NOT IN (SELECT * FROM t_filter)`. There is also a column mask applied to `t_base.b` with the expression `IF(b IN SELECT * FROM t_mask, b, NULL)`.

The `TableInfo` for `t_base` will contain:
```
filters -> ["a NOT IN (SELECT * FROM t_filter)"]
columns -> ["a", empty], ["b", "IF(b IN SELECT * FROM t_mask, b, NULL)"]
referenceChain -> []
```

The `TableInfo` for `t_filter` will contain:
```
filters -> []
columns -> []
referenceChain -> [ROW_FILTER, "a NOT IN (SELECT * FROM t_filter)", "default", "db", "t_base"]
```

The `TableInfo` for `t_mask` will contain:
```
filters -> []
columns -> []
referenceChain -> [COLUMN_MASK, "IF(b IN SELECT * FROM t_mask, b, NULL)", "default", "db", "t_base"]
```

Is the difference clear now? The `filters`/`columns` fields on `TableInfo` are more like understanding how a given table scan was impacted; whereas the filter/mask info in `referenceChain` is more like understanding what reference caused the table to be included in the plan.